solutions/go/ping-pong-multiple/code/app/server.go

```diff
@@ -22,11 +24,23 @@ func main() {
 	defer conn.Close()
 
 	for {
-		if _, err := conn.Read([]byte{}); err != nil {
-			fmt.Println("Error reading from client: ", err.Error())
-			continue
+		buf := make([]byte, 1024)
+
+		if _, err := conn.Read(buf); err != nil {
+			if err == io.EOF {
+				break
+			} else {
+				fmt.Println("error reading from client: ", err.Error())
+				os.Exit(1)
+			}
 		}
 
-		conn.Write([]byte("+PONG\r\n"))
+		// Let's use a simple substring check for now. We'll implement a proper Redis Protocol parser in later stages.
+		if bytes.Contains(buf, []byte("ping")) {
```

Comment:
I'm not sure if case-sensitivity is checked in further steps, but take into account that the redis server doesn't care about it. If you send `ping or PING or PinG`, it'll answer correctly. We're not validating this here.

Reply:
Fair point! Doing this with proper case-insensitivity will be unnecessarily complicated at this stage, I've proposed an alternative that does away with this check altogether [here](https://github.com/codecrafters-io/build-your-own-redis/pull/49#discussion_r1016701192).